clap_blocks/src/compactor2.rs

```diff
@@ -244,4 +244,14 @@ pub struct Compactor2Config {
         action
     )]
     pub process_all_partitions: bool,
+
+    /// Maximum number of columns in the table of a partition that will be able to considered
+    /// to get compacted
+    #[clap(
+        long = "compaction-max-num-columns-per-table",
+        env = "INFLUXDB_IOX_COMPACTION_MIN_NUM_COLUMNS_PER_TABLE",
+        default_value = "200",
+        action
+    )]
+    pub max_num_columns_per_table: usize,
```

Comment:
I think that when we were talking about this you made the argument that we **should not** enable this by default because we will skip too much data - we should only enable it as a diagnostic to understand the OOMs - is that right? It makes sense to me!

If so, we should make this filter _optional_ and disabled by default - so we can turn it on when needed, but otherwise it's unused / doesn't affect the normal compaction.

Perhaps we make this optional (ideally, because it makes for nice CLI):

```suggestion
        action
    )]
    pub max_num_columns_per_table: Option<usize>,
```

Or set it to an impossibly high default (effectively disabling it) if not:

```suggestion
        default_value = "10000",
        action
    )]
    pub max_num_columns_per_table: usize,
```

Then we enable it in the one env we want to use to test our hypothesis via config :+1:

Reply:
I make it 10000 as default. After this is merged, I will make it 200 on mirror cluster for us to observe  the behavior